rerun_py/rerun_sdk/rerun/time.py

```diff
@@ -108,35 +108,44 @@ def set_time(
         )
 
 
-def to_nanos(duration: int | float | timedelta | np.timedelta64) -> int:
-    if isinstance(duration, (int, np.integer)):
-        return 1_000_000_000 * int(duration)  # Interpret as seconds and convert to nanos
-    elif isinstance(duration, float):
-        return round(1e9 * duration)  # Interpret as seconds and convert to nanos
+def to_nanos(duration: int | np.integer | float | np.float64 | timedelta | np.timedelta64) -> int:
+    if isinstance(duration, np.timedelta64):
+        return duration.astype("timedelta64[ns]").astype("int64")  # type: ignore[no-any-return]
     elif isinstance(duration, timedelta):
         return round(1e9 * duration.total_seconds())
-    elif isinstance(duration, np.timedelta64):
-        return duration.astype("timedelta64[ns]").astype("int64")  # type: ignore[no-any-return]
+    elif isinstance(duration, (int, np.integer)):
+        return 1_000_000_000 * int(duration)  # Interpret as seconds and convert to nanos
+    elif isinstance(
+        duration,
+        # Only allowing doubles since 32-bit and 16-bit floats lose precision when multiplying at this scale
+        (float, np.float64),
```

Comment:
I think this should be up to the user.

For instance: f32 still has millisecond precision for durations under ~4 hours. Not great; not terrible.